src/database/remove.ts

```diff
@@ -0,0 +1,100 @@
+import * as pathLib from "path";
+import { Queue } from "../queue";
+import * as FirebaseError from "../error";
+import * as logger from "../logger";
+import RemoveRemote, { PrefetchResult } from "./remove-remote";
+
+export interface DatabaseRemoveOptions {
+  // RTBD instance ID.
+  instance: string;
+  // Number of concurrent chunk deletes to allow.
+  concurrency: number;
+  // Number of retries for each chunk delete.
+  retries: number;
+}
+
+export default class DatabaseRemove {
+  public path: string;
+  public concurrency: number;
+  public retries: number;
+  public remote: any;
+  private jobQueue: Queue<string>;
+  private waitingPath: Map<string, number>;
+
+  /**
+   * Construct a new RTDB delete operation.
+   *
+   * @constructor
+   * @param path path to delete.
+   * @param the RTDB instance ID.
+   * @param options.concurrency the number of concurrent chunk delete allowed
+   * @param options.retires the number of retries for each chunk delete
```

Comment:
I think these `@param`s are stale. How about

```typescript
* @param path path to delete.
* @param options
```

The rest of `options`' documentation is in the `DatabaseRemoveOptions` interface, so individual keys don't need to be documented.